server/parser.go

```diff
@@ -729,21 +858,14 @@ func protoSnippet(start int, buf []byte) string {
 // clonePubArg is used when the split buffer scenario has the pubArg in the existing read buffer, but
 // we need to hold onto it into the next read.
 func (c *client) clonePubArg() {
+	// Just copy and re-process original arg buffer.
 	c.argBuf = c.scratch[:0]
-	c.argBuf = append(c.argBuf, c.pa.subject...)
-	c.argBuf = append(c.argBuf, c.pa.reply...)
-	c.argBuf = append(c.argBuf, c.pa.sid...)
-	c.argBuf = append(c.argBuf, c.pa.szb...)
-
-	c.pa.subject = c.argBuf[:len(c.pa.subject)]
+	c.argBuf = append(c.argBuf, c.pa.arg...)
 
-	if c.pa.reply != nil {
-		c.pa.reply = c.argBuf[len(c.pa.subject) : len(c.pa.subject)+len(c.pa.reply)]
+	// This is a routed msg
+	if c.pa.account != nil {
+		c.processRoutedMsgArgs(c.argBuf)
+	} else {
+		c.processPub(c.argBuf)
```

Comment:
One consequence of this is that now message is traced twice (because `processPub` logs the message once entering the function).

https://github.com/nats-io/gnatsd/blob/d33be3aaa3d4f7dc1bee2b582927c99a6bfd8d96/server/client.go#L1208-L1211

```sh
[4443] 2018/10/24 22:10:09.013848 [DBG] 127.0.0.1:50851 - cid:3 - Client connection created
[4443] 2018/10/24 22:10:12.182430 [TRC] 127.0.0.1:50851 - cid:3 - <<- [PUB hello 5]
[4443] 2018/10/24 22:10:12.182448 [TRC] 127.0.0.1:50851 - cid:3 - <<- [PUB hello 5]
[4443] 2018/10/24 22:10:13.588301 [TRC] 127.0.0.1:50851 - cid:3 - <<- MSG_PAYLOAD: [world]
```

Reply:
You mean the way we handle the split buffer correct?

Reply:
Got it, will fix.